### PR TITLE
Base scss hugo structure

### LINKED ISSUE #24: Organize SCSS structure for the Hugo theme
## User Story

_As a Hugo developer,_
_I want my `.scss` files to be organized,_
_so we'll avoid mess and have a well-structured project outline._

## Requirements

* Think about the well-structured outline for style files specifically _for Hugo_ sites/themes.
I encourage you to check how other people achieve it for their Hugo projects. You can also seek for non-Hugo sites structure for the ideas.
* Create `.scss` style files even if they are empty to show how your folder/files structure will look like.
* Rewrite [`styles.css`](https://github.com/animegungrave/hugo-knowledge-base-theme/pull/22/files#diff-a16a51aaea7dc21d3202074fc7228024) from #22 pull request to well-aligned `.scss` format.

## Open Questions

- [x] Where are we going to store `.scss` files for Hugo theme?
  - [x] Is `static` folder good enough for this purpose? - No, [as it's stated](https://gohugo.io/content-management/static-files/), these files cannot be processed.
  - [x] What about [`assets`](https://github.com/alex-shpak/hugo-book/tree/1941d71/assets) folder?
  - [ ] Can we store `.scss` along with the target `.html` template like Angular projects do? Do we need this approach really?
- [x] Is it a good idea to have `.scss` per template (per `single.html`, `list.html`, `baseof.html`, any partial template, etc.) or not?

### LINKED ISSUE #30: Add .gitignore with common settings
## Requirements
* Create `.gitignore` file in the root of the repository
* Cofigure base settings

Additional [stuff](https://github.com/github/gitignore) as example

### PR DESCRIPTION
**Pull request by alexander-derevyanko**
Closes #24
Closes #30

It's just a basic `scss` folders structure for further development.

Additional stuff: [one](https://matthewelsom.com/blog/simple-scss-playbook.html) and [two](https://www.sitepoint.com/architecture-sass-project/)

### CODE REVIEW

**Comment by alexander-derevyanko** [comment]
We have to use HUGO `extended` version for preventing issues during transformation.
Please pay attention on [this](https://prnt.sc/q7v7d9)
Additional [link](https://gohugo.io/troubleshooting/faq/#i-get-this-feature-is-not-available-in-your-current-hugo-version)

**Comment by YuriyOborozhnyi** [comment]
Hey @alexander-derevyanko, could you please adjust the description of your PR in order to fix issues links?

They are missing spaces between "Fixes" and issue's number.

You can also check [this link](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords) for the curiosity.

**Comment by YuriyOborozhnyi** [comment]
@alexander-derevyanko, let me merge #22 PR first, so we'll have the initial layout. Making the changes you requested.

**Comment by alexander-derevyanko** [comment]
@YuriyOborozhnyi sure. I planned it that way

**Comment by YuriyOborozhnyi** [comment]
@alexander-derevyanko, cool. I'd recommend to create "draft" pull request for this purpose (if you are not going to merge them yet).

**Comment by YuriyOborozhnyi** [comment]
> We have to use HUGO `extended` version for preventing issues during transformation.
> Additional [link](https://gohugo.io/troubleshooting/faq/#i-get-this-feature-is-not-available-in-your-current-hugo-version) - last point

Correct. However, we should be able to give users the ability to use our theme [without installing Hugo extended version](https://discourse.gohugo.io/t/should-i-use-hugo-extended-for-a-new-hugo-project/13954/6) on their own machines.

**Comment by YuriyOborozhnyi** [comment]
@alexander-derevyanko, Render build for PR doesn't look good.
https://hugo-knowledge-base-theme-pr-33.onrender.com/

I assume it's because of the extended Hugo version (we are not using it on Render). I would encourage you to _not_ change the Hugo version on Render, but to try to use the approach with resources I mentioned above.

You can also see that [Hugo Book theme is using this approach as well](https://github.com/alex-shpak/hugo-book/tree/master/exampleSite/resources/_gen/assets/scss).

image

As I can see from [the forum's discussion](https://discourse.gohugo.io/t/hugo-extended-needed-for-hugo-fresh-theme/13844/4), this is also kinda "must":

> We have a policy in which Hugo themes assets should be considered as stable and theme authors who use Hugo Extended features should commit their /resources/ directory so that a theme works in all versions of Hugo.

File: exampleSite/resources/_gen/assets/scss/stylesheets/main.scss_48b060fe05b0a273d182ef83c0605941.content [hunk 1]
```diff
@@ -0,0 +1 @@
+html,body{font-family:Segoe UI,SegoeUI,Arial,sans-serif;height:100%}body{display:flex;flex-direction:column;padding:0;margin:0}.border{border:1px solid red}.tree-pane{width:25%}main{width:50%}.main-container{display:flex;flex:1 0 auto}.toc{width:25%}footer{flex-shrink:0}
```
Comment (YuriyOborozhnyi) [inline exampleSite/resources/_gen/assets/scss/stylesheets/main.scss_48b060fe05b0a273d182ef83c0605941.content]: I think we can remove these files from Git. They are generated each build as I can see, so we have no need to track them. They're artifacts. See [this forum's comment](https://discourse.gohugo.io/t/why-does-hugo-create-the-resources-directory/13021/4) as well.

I propose to add `resources/_gen/` to `.gitignore`.

**Comment by YuriyOborozhnyi** [comment]
@alexander-derevyanko, also please take a look at [my comment](https://github.com/animegungrave/hugo-knowledge-base-theme/issues/24#issuecomment-564235818) for the corresponding ticket. It is related to this PR.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# See http://help.github.com/ignore-files/ for more about ignoring files.
+
+# compiled output
```
Comment (YuriyOborozhnyi) [inline .gitignore]: Are these files from `# compiled output` Hugo-specific or you found them somewhere else?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+/.sass-cache
+/libpeerconnection.log
+npm-debug.log
+testem.log
+package-lock.json
+
+
+# System Files
+.DS_Store
+Thumbs.db
```
Comment (YuriyOborozhnyi) [inline .gitignore]: Also please consider installing Editor Config for VS Code extension and fix the missing new line at the end of the file.

File: layouts/_default/baseof.html [hunk 4]
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE html>
 <html lang="{{ .Site.Language.Lang }}">
-    {{ partial "head.html" . }}
+    <head>
```
[the remark this answers is not in the document]
Comment (alexander-derevyanko) [reply layouts/_default/baseof.html]: No, I don't think so. `baseof.html` is an entry point of the structure and `<head>` is one of the main tag. `head.html` just represents the insides of the `<head>`.

File: exampleSite/resources/_gen/assets/scss/stylesheets/main.scss_48b060fe05b0a273d182ef83c0605941.content (hunk 1 above)
Comment (alexander-derevyanko) [reply exampleSite/resources/_gen/assets/scss/stylesheets/main.scss_48b060fe05b0a273d182ef83c0605941.content]: As we discussed earlier, we'd provide`resources/_get/` to Git.
Please, take a look: [one](https://discourse.gohugo.io/t/should-i-use-hugo-extended-for-a-new-hugo-project/13954/6), [two](https://github.com/alex-shpak/hugo-book/tree/master/exampleSite/resources/_gen/assets/scss), [three](https://discourse.gohugo.io/t/hugo-extended-needed-for-hugo-fresh-theme/13844/4)

File: .gitignore (hunk 2 above)
Comment (alexander-derevyanko) [reply .gitignore]: I've tried to create generic `.gitignore` file without Hugo-specific cases.

**Comment by alexander-derevyanko** [comment]
> @alexander-derevyanko, also please take a look at [my comment](https://github.com/animegungrave/hugo-knowledge-base-theme/issues/24#issuecomment-564235818) for the corresponding ticket. It is related to this PR.

@YuriyOborozhnyi take a look, please

File: exampleSite/resources/_gen/assets/scss/scss/main.scss_48b060fe05b0a273d182ef83c0605941.content [hunk 5]
```diff
@@ -0,0 +1 @@
+html,body{font-family:Segoe UI,SegoeUI,Arial,sans-serif;height:100%}body{display:flex;flex-direction:column;padding:0;margin:0}.border{border:1px solid red}.tree-pane{width:25%}main{width:50%}.main-container{display:flex;flex:1 0 auto}.toc{width:25%}footer{flex-shrink:0}
```
Comment (YuriyOborozhnyi) [inline exampleSite/resources/_gen/assets/scss/scss/main.scss_48b060fe05b0a273d182ef83c0605941.content]: Let's add these generated files (`resources/_gen`) _for now_.

File: assets/scss/layout/partials/_main-block.scss [hunk 6]
```diff
@@ -0,0 +1,8 @@
+main {
```
Comment (YuriyOborozhnyi) [inline assets/scss/layout/partials/_main-block.scss]: It should be a part of `/assets/scss/layouts/partials/_baseof.scss`, not `_main-block.scss`.

**Pull request merged**